src/index.ts

```diff
@@ -418,11 +423,26 @@ export const newForm: NewFormFn = <T extends object>(
       [field]: value,
     }));
 
+    const isDirtyNow = checkIsDirty();
+    __isDirty.set(isDirtyNow);
+
     if (shouldValidateField && config.validationSchema) {
       validateFieldSync(field);
     }
   };
 
+  const checkIsDirty = (): boolean => {
+    const value = get(values);
+    const inititial_values = get(__initialValues);
```

Comment:
Use camel case please

```suggestion
    const initialValues = get(__initialValues);
```